### PR TITLE
Improve Automation Coverage for Critical path of Android

### PR DESCRIPTION
**Pull request by bilalawan321**
[LEARNER-8407](https://openedx.atlassian.net/browse/LEARNER-8407)

Following scenarios is implemented:

- Update strings for login and new landing screen
- Add test cases for reset password and wrong credentials in the login screen
- Verify that, if the user fills all data except "Full Name" and tap on the 'Create my account' button,
     an alert pop-up should appear along with the error message.
- Verify that, if the user fills all data except "User Name" and tap on the 'Create my account' button,
     an alert pop-up should appear along with the error message.
- Verify that, if the user fills all data except "Password" and tap on the 'Create my account' button,
      an alert pop-up should appear along with the error message.
- Verify that, if the user fills all data except "Country" and tap on the 'Create my account' button,
       an alert pop-up should appear along with the error message.
- Verify that, if user fill all data but with wrong email format and tap on 'Create my account' button,
      email format error message should appear.
- Verify that, if user fill all data but with wrong email format and tap on 'Create my account' button,
     email format error message should appear.
- Verify that, password required at least 8 characters with a special character in it.
- Verify on tapping Programs will load Programs contents in its tab
- Verify on tapping Programs will load Programs contents in its tab in landscape mode
- Updated scenarios string for full profile view
- Updated test case of limited profile view
- Verify user can update his location
- Verify user can update his language
- Verify user can update his about me info
- Update doc strings for edit profile, after adding test cases of language,location and about me info updation
- Add test case of logout feature in course_html_component screen
- Add test case of logout feature in course_subsection screen
- Add edit_profile, video_subsection and video_dashboard screen in android full execution cycle command
- Remove coding=utf-8 from android pages and tests files

### CODE REVIEW

File: tests/android/tests/test_android_edit_profile.py [hunk 1]
```diff
@@ -171,6 +169,60 @@ def test_profile_views_smoke(self, set_capabilities, setup_logging):
         assert edit_profile_screen.get_by_class_from_elements(
             android_elements.all_textviews,
             global_contents.ninth_existence).get_attribute('enabled') == 'false'
+
+    def test_update_profile_info(self, set_capabilities, setup_logging):
```
Comment (jawad-khan) [inline tests/android/tests/test_android_edit_profile.py:173]: can we break this function?

File: tests/android/tests/test_android_register.py [hunk 2]
```diff
@@ -146,6 +159,65 @@ def test_required_and_optional_fields_smoke(self, set_capabilities, setup_loggin
         assert android_register_page.get_password_validation_textview().text == strings.REGISTER_PASSWORD_BLANK_ERROR
         assert android_register_page.get_country_validation_textview().text == strings.REGISTER_COUNTRY_BLANK_ERROR
 
+        user_name = global_contents.generate_random_credentials(5)
+        email = user_name + '@example.com'
+        first_name = global_contents.generate_random_credentials(4)
+        last_name = global_contents.generate_random_credentials(4)
+        name = first_name + ' ' + last_name
```
Comment (jawad-khan) [inline tests/android/tests/test_android_register.py:166]: use string formatting instead of +.

Comment (bilalawan321) [reply tests/android/tests/test_android_register.py]: fixed

File: tests/android/tests/test_android_edit_profile.py (hunk 1 above)
Comment (bilalawan321) [reply tests/android/tests/test_android_edit_profile.py]: fixed

**Comment by bilalawan321** [comment]
@jawad-khan I have fixed your mentioned points, can you please verify?

**Review by jawad-khan — APPROVED** [review]

**Pull request merged**